ktor-retrofit/src/main/kotlin/com/bnorm/ktor/retrofit/RetrofitService.kt

```diff
@@ -170,6 +186,23 @@ private fun Route.process(service: Any, function: KFunction<*>) {
   TODO("implement the rest of the function annotations : $annotations")
 }
 
+private fun Route.checkRegisteredRoute(
+  method: HttpMethod,
+  baseUrl: Route,
+  path: String,
+  serviceInterface: KClass<*>,
+  methodName: String
+) {
+  val registeredRoutes = attributes[injectorKey]
+  val newRoute = RegisteredRoute(method, baseUrl.toString(), path, serviceInterface.qualifiedName, methodName)
+  val registeredPaths = registeredRoutes.getOrPut(method) { mutableSetOf() }
+  val registeredPath = registeredPaths.firstOrNull { registeredPath -> registeredPath == newRoute }
```

Comment:
So fun insight, a LinkedHashSet uses a LinkedHashMap in the background for storage. Maybe use a Map directly instead of a Set with the RegisteredRoute as the key and value so you can call `get` on the map and not navigate the entire set to find the previous value?

Reply:
I think I understand, pushed up to use a map instead